src/robot/running/namespace.py

```diff
@@ -338,6 +339,42 @@ def _get_runner_from_libraries(self, name):
             return found[0]
         self._raise_multiple_keywords_found(name, found)
 
+    def _handle_private_runners(self, runners):
+        sum_public = sum("robot:private" not in runner.tags for runner in runners)
+        sum_private = sum("robot:private" in runner.tags for runner in runners)
+
+        if sum_public == len(runners) or sum_private == len(runners) or sum_public > 1:
```

Comment:
The first condition doesn't seem to be necessary. We only get to this function if there were multiple matches, so just having `sum_public > 1` ought to be enough to cover also the `sum_public == len(runners)` case.

We generally haven't used empty lines between functions. If there's a need to use them for grouping code, we have typically created helper functions instead. That could be considered hare as well or possibly the empty lines could just be removed.